entgql/extension.go

```diff
@@ -231,6 +285,299 @@ func (e *Extension) mapScalar(f *gen.Field, op gen.Op) string {
 	return scalar
 }
 
+// genSchema generates the graphql schema
+func (e *Extension) genSchema() gen.Hook {
+	return func(next gen.Generator) gen.Generator {
+		var (
+			queries    = make(map[string]*ast.FieldDefinition)
+			interfaces = make(map[string]*ast.InterfaceDefinition)
+			scalars    = make(map[string]*ast.ScalarDefinition)
+			objects    = make(map[string]*ast.ObjectDefinition)
+			enums      = make(map[string]*ast.EnumDefinition)
+			inputs     = make(map[string]*ast.InputObjectDefinition)
+		)
+		return gen.GenerateFunc(func(g *gen.Graph) error {
+			nodes, err := filterNodes(g.Nodes)
+			if err != nil {
+				return err
+			}
+			if err := next.Generate(g); err != nil {
+				return err
+			}
+
+			addTimeScalarDefinition(scalars)
+			addNodeDefinition(interfaces)
+			// Iterate through the nodes and create the base objects, enums and queries
+			for _, node := range nodes {
+				e.addEnumDefinitions(enums, node)
+				e.addObjectDefinition(objects, node)
+
+				var hasOrderByType bool
+				// If orderBy has been enabled, create the types for ordering
+				if e.orderBy {
+					hasOrderByType = e.addOrderByDefinitions(enums, inputs, node)
+				}
+
+				if e.whereFilters {
+					// Create the where input types
+					err := e.addWhereType(inputs, node)
+					if err != nil {
+						return err
+					}
+				}
+
+				if e.relaySpec {
+					e.addRelayObjectDefinitions(objects, node)
+					e.addQueryDefinition(queries, node, true, hasOrderByType)
```

Comment:
This you should probably leave up to the user to define. For example If they use custom filters. Or make it an opt in annotation